source/common/common/token_bucket_impl.cc

```diff
@@ -4,9 +4,14 @@
 
 namespace Envoy {
 
+namespace {
+// The minimal fill rate will be one second every year.
+constexpr double kMinFillRate = 1.0 / (365 * 24 * 60 * 60);
+} // namespace
+
 TokenBucketImpl::TokenBucketImpl(uint64_t max_tokens, TimeSource& time_source, double fill_rate)
-    : max_tokens_(max_tokens), fill_rate_(std::abs(fill_rate)), tokens_(max_tokens),
```

Comment:
Actually, in implementation, it seems this is more generic than just xDS?

Reply:
The implementation is indeed more generic.
The other use-cases where I see this is used are via these config knobs (which from API standpoint may be unfortunate):
- [BandwidthLimit](https://github.com/envoyproxy/envoy/blob/9c1dcef98fe3982e26ec9e24ccd19de39862949d/api/envoy/extensions/filters/http/bandwidth_limit/v3/bandwidth_limit.proto#L62) - the lower bound there is higher then the one we propose for xDS.
- [Redis Proxy](https://github.com/envoyproxy/envoy/blob/9c1dcef98fe3982e26ec9e24ccd19de39862949d/api/envoy/extensions/filters/network/redis_proxy/v3/redis_proxy.proto#L233) - uses an unsigned integer value, and so should be higher than the proposed lower bound.

Reply:
Ack for BandwidthLimit, but Redis proxy could be zero today right?

Reply:
I think it can't be 0, because of [this line](https://github.com/envoyproxy/envoy/blob/f9ffd76e1b6982666d2ac0556fde441357d25341/source/common/common/token_bucket_impl.cc#L45):
```
return std::chrono::milliseconds(static_cast<uint64_t>(std::ceil((1 / fill_rate_) * 1000)));
```
which will cause division by 0.

It seems that the Redis code guards against no-value:
https://github.com/envoyproxy/envoy/blob/f9ffd76e1b6982666d2ac0556fde441357d25341/source/extensions/filters/network/redis_proxy/conn_pool_impl.cc#L249-L253

but it may make sense to add a PGV there to ensure it is above 0. WDYT?

Reply:
Apologies for the delay. Added the PGV constraint, and updated the release notes.